### PR TITLE
Change DeformationFieldTransform and PositionFieldTransform to use Compsite<T>

### PR DESCRIPTION
**Pull request by cmhulbert**

### CODE REVIEW

**Comment by cmhulbert** [comment]
@bogovicj let me know what you think. It's in draft for now, but I think I implemented everything in #41

**Review by bogovicj — CHANGES_REQUESTED** [review]
A couple minor things + we may want to change the implementation in places.

File: src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java [hunk 1]
```diff
@@ -52,66 +55,54 @@
...
 	public void apply( final RealLocalizable source, final RealPositionable target )
 	{
-		for ( int d = 0; d < positionAccesses.length; d++ )
-			positionAccesses[ d ].setPosition( source );
 
-		for ( int d = 0; d < positionAccesses.length; d++ )
-			target.setPosition( positionAccesses[ d ].get().getRealDouble() + source.getDoublePosition( d ), d );
+		Composite< T > comp = positionAccesses.setPositionAndGet(source);
+		for ( int d = 0; d < target.numDimensions(); d++ )
+			target.setPosition( comp.get( d ).getRealDouble() + source.getDoublePosition( d ), d );
```
[the remark this answers is not in the document]
Comment (axtimwalde) [reply src/main/java/net/imglib2/realtransform/DeformationFieldTransform.java]: Good point!  Done.

File: src/main/java/net/imglib2/realtransform/PositionFieldTransform.java [hunk 2]
```diff
@@ -57,105 +53,77 @@
...
 	@Override
 	public void apply( final double[] source, final double[] target )
 	{
-		for ( int d = 0; d < positionAccesses.length; d++ )
-			positionAccesses[ d ].setPosition( source );
 
-		for ( int d = 0; d < positionAccesses.length; d++ )
-			target[ d ] = positionAccesses[ d ].get().getRealDouble();
+		RealComposite<T> t = positionAccesses.setPositionAndGet(source);
+		for ( int d = 0; d < target.length; d++ )
```
[the remark this answers is not in the document]
Comment (axtimwalde) [reply src/main/java/net/imglib2/realtransform/PositionFieldTransform.java]: Nice! Done.

File: src/main/java/net/imglib2/realtransform/PositionFieldTransform.java [hunk 3]
```diff
@@ -57,105 +53,77 @@
 public class PositionFieldTransform< T extends RealType< T > > implements RealTransform
 {
 	/* one for each dimension */
-	protected final RealRandomAccess< T >[] positionAccesses;
+	protected final RealRandomAccess< RealComposite< T > > positionAccesses;
```
[the remark this answers is not in the document]
Comment (axtimwalde) [reply src/main/java/net/imglib2/realtransform/PositionFieldTransform.java]: Or access because inherited by displacementfields, done.

**Comment by bogovicj** [comment]
I'd like to add some tests that check

* `PositionFieldTransform`s with  `numSourceDimensions() != numTargetDimensions()`
* applying these transforms for points of higher dimensionality

**Review by bogovicj — APPROVED** [review]

**Pull request merged**